src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1275,12 +1275,21 @@ private ICurrentSessionContext BuildCurrentSessionContext()
 					return new ThreadStaticSessionContext(this);
 				case "web":
 					return new WebSessionContext(this);
-				case "wcf_operation":
-					return new WcfOperationSessionContext(this);
 			}
 
 			try
 			{
+				// The WcfOperationSessionContext type that was previously included in this
+				// assembly is now in an optional external assembly, so to maintain
+				// compatibility we map the old alias and type names to the updated type name.
+				const string wcfTypeName = "NHibernate.Context.WcfOperationSessionContext";
+				if (impl.Equals("wcf_operation", StringComparison.Ordinal) ||
+					impl.StartsWith(wcfTypeName, StringComparison.OrdinalIgnoreCase))
+				{
+					var wcfAssemblyName = typeof(SessionFactoryImpl).Assembly.FullName.Replace("NHibernate,", "NHibernate.Wcf,");
+					impl = string.Concat(wcfTypeName, ", ", wcfAssemblyName);
+				}
```

Comment:
We likely need a dedicated error handling for this case, at least for hinting at referencing the new library. Maybe should we add something like `isWcfOperationContext = true` here then test in the catch for this condition in order to add in the error message something like `(Is a reference to NHibernate.Wcf assembly missing?)`.